lib/collection/property.js

```diff
@@ -178,12 +187,12 @@ _.extend(Property, /** @lends Property */ {
         // extract all arguments to get the list of variables
         !_.isArray(variables) && (variables = _.tail(arguments));
 
-        return str.replace(this.REGEX_EXTRACT_VARS, function (match, token) {
-            var r = _.findValue(variables, token);
-            r && _.isFunction(r) && (r = r());
-            r && _.isFunction(r.toString) && (r = r.toString());
-            return nativeTypes[(typeof r)] ? r : match;
-        });
+        while (resolvedStr != str && ++loopCounter < 20) {
```

Comment:
1. Given that a fixed limit of 20 levels of depth checking has to be performed, a loop would be a better choice. This can be shifted to recursion with a fair amount of TCO, but a loop is more readable here.

2. I'm not quite sure how else the check for any remaining possible substitutions would work.